dig/README.md

```diff
@@ -0,0 +1,78 @@
+# Dependency Injection Graph
+
+This package provides a fairly opinionated way of resolving object dependencies.
+There are two sides of dig: `Inject` and `Resolve`
+
+## Inject
+
+Adding an object, or a constructor of an object to the graph.
+
+There are two ways of providing an object:
+
+1. Injecting a pointer to an existing object
+1. Injecting a "constructor function" that returns one pointer (or interface)
+
+### Injecting an object
+
+Injecting an object means it has no dependencies, and will be used as a
+**shared** singleton instance for all resolutions within the graph
+
+```go
+type Fake struct {
+    Name string
+}
+err := g.Inject(&Fake{Name: "I am an injected thing"})
```

Comment:
This could be a 3 step process. I like the term `Register` then `Resolve` graph, and `Inject`. In think `Inject` is used when object is ready to be consumed, and gets 'Injected' in the code. That way may be we can hide the Resolve call from users?
 
1. Graph is registered with all the dependency objects
2. DIG resolves the dependency graph
3. Objects are Injected in the code